icu4c/source/i18n/formattedval_sbimpl.cpp

```diff
@@ -219,12 +219,19 @@ bool FormattedValueStringBuilderImpl::nextPositionImpl(ConstrainedFieldPosition&
     }
 
     U_ASSERT(currField == kUndefinedField);
+    // Always set the position to the end so that we don't revisit previous sections
+    cfpos.setState(
+        cfpos.getCategory(),
+        cfpos.getField(),
+        fString.fLength,
+        fString.fLength);
     return false;
 }
 
 void FormattedValueStringBuilderImpl::appendSpanInfo(int32_t spanValue, int32_t length) {
-    if (spanIndices.getCapacity() <= spanValue) {
-        spanIndices.resize(spanValue * 2);
+    U_ASSERT(spanIndices.getCapacity() >= spanValue);
```

Comment:
Should this be > only?

Reply:
if you keep this as >= in the U_ASSERT then I think your debug build will bark when you run the test when spaceValue is 8

Reply:
spanValue is never greater than getCapacity. The <= before was misleading; it should have always been ==. I added the U_ASSERT to emphasize that it's safe and equivalent to replace the <= with a ==.

Please note that the comparison is not always in the same order; sometimes it is A-B and other times B-A.

Reply:
sorry, I must be too tired when I review your code last night. Yea you are right